### ISSUE TITLE
Crashed during / after cleanup

### ISSUE THREAD

**Issue by freakytoad1** [louislam/uptime-kuma#1520]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 👟 Reproduction steps

At 3:14am, it looks like the 'clear-old-data' job happened. Immediately after an error was thrown and the container stopped. I did not have 'restart=unless-stopped' on so it was down until 9:11pm when I noticed and restarted the container.

### 👀 Expected behavior

That a crash does not occur

### 😓 Actual Behavior

Crash occurred that appears to be related to 'clear-old-data' job.

### 🐻 Uptime-Kuma Version

1.14.0

### 💻 Operating System and Arch

Docker on Unraid

### 🌐 Browser

Firefox

### 🐋 Docker Version

20.10.5

### 📝 Relevant log output

```shell
today at 3:13:46 AMMonitor #22 'unifi': Successful Response: 22 ms | Interval: 60 seconds | Type: http
today at 3:14:00 AMWorker for job "clear-old-data" online undefined
today at 3:14:00 AMData Dir: data/
today at 3:14:00 AM(node:7) Warning: Accessing non-existent property 'Proxy' of module exports inside circular dependency
today at 3:14:00 AM(Use `node --trace-warnings ...` to show where the warning was created)
today at 3:14:00 AMWelcome to Uptime Kuma
today at 3:14:00 AMYour Node.js version: 16
today at 3:14:00 AMNode Env: production
today at 3:14:00 AMImporting Node libraries
today at 3:14:00 AMImporting 3rd-party libraries
today at 3:14:00 AMImporting this project modules
today at 3:14:00 AMPrepare Notification Providers
today at 3:14:00 AMVersion: 1.14.0
today at 3:14:00 AMCreating express and socket.io instance
today at 3:14:00 AMServer Type: HTTP
today at 3:14:00 AMData Dir: ./data/
today at 3:14:00 AMConnecting to the Database
today at 3:14:00 AM./server/model/monitor.js is not a valid BeanModel, skipped
today at 3:14:00 AM(node:7) Warning: Accessing non-existent property 'Proxy' of module exports inside circular dependency
today at 3:14:00 AM(node:7) Warning: Accessing non-existent property 'prototype' of module exports inside circular dependency
today at 3:14:00 AMSQLite config:
today at 3:14:00 AMSQLite config:
today at 3:14:00 AM[ { journal_mode: 'wal' } ]
today at 3:14:00 AM[ { journal_mode: 'wal' } ]
today at 3:14:00 AM[ { cache_size: -12000 } ]
today at 3:14:00 AM[ { cache_size: -12000 } ]
today at 3:14:00 AMSQLite Version: 3.36.0
today at 3:14:00 AMConnected
today at 3:14:00 AMSQLite Version: 3.36.0
today at 3:14:00 AMYour database version: 10
today at 3:14:00 AMLatest database version: 10
today at 3:14:00 AMDatabase patch not needed
today at 3:14:00 AMDatabase Patch 2.0 Process
today at 3:14:00 AM[Background Job]: {
today at 3:14:00 AM  name: 'clear-old-data',
today at 3:14:00 AM  message: 'Clearing Data older than 30 days...'
today at 3:14:00 AM}
today at 3:14:00 AM[Background Job]: { name: 'clear-old-data', message: 'done' }
today at 3:14:00 AMFATAL ERROR: Error::ThrowAsJavaScriptException napi_throw
today at 3:14:00 AM 1: 0xb09980 node::Abort() [node]
today at 3:14:00 AM 2: 0xa1c235 node::FatalError(char const*, char const*) [node]
today at 3:14:00 AM 3: 0xa1c23e  [node]
today at 3:14:00 AM 4: 0xad139b napi_fatal_error [node]
today at 3:14:00 AM 5: 0x1457dde7c556  [/app/node_modules/@louislam/sqlite3/lib/binding/napi-v6-linux-x64/node_sqlite3.node]
today at 3:14:00 AM 6: 0x1457dde7c9d8 Napi::Error::ThrowAsJavaScriptException() const [/app/node_modules/@louislam/sqlite3/lib/binding/napi-v6-linux-x64/node_sqlite3.node]
today at 3:14:00 AM 7: 0x1457dde916ae node_sqlite3::Statement::Statement(Napi::CallbackInfo const&) [/app/node_modules/@louislam/sqlite3/lib/binding/napi-v6-linux-x64/node_sqlite3.node]
today at 3:14:00 AM 8: 0x1457dde96885 Napi::ObjectWrap<node_sqlite3::Statement>::ConstructorCallbackWrapper(napi_env__*, napi_callback_info__*) [/app/node_modules/@louislam/sqlite3/lib/binding/napi-v6-linux-x64/node_sqlite3.node]
today at 3:14:00 AM 9: 0xab44dd  [node]
today at 3:14:00 AM10: 0xd54c3e  [node]
today at 3:14:00 AM11: 0xd55207 v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [node]
today at 3:14:00 AM12: 0x15f0bf9  [node]
today at 9:11:35 PMContainer stopped
```

**Comment by louislam** [comment]
It looks like it still hit by the worker_threads issue of node_sqlite3,

https://github.com/TryGhost/node-sqlite3/pull/1367

I updated node-sqlite3 library in the upcoming version of Uptime Kuma, hopefully will fix this issue.

**Comment by louislam** [comment]
Tried node-sqlite3 5.0.3 and 5.0.4, it is still happening.

**Comment by Rustymage** [comment]
Just happened on my instance

```
Worker for job "clear-old-data" online undefined
Data Dir: data/
(node:7) Warning: Accessing non-existent property 'Proxy' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
Welcome to Uptime Kuma
Your Node.js version: 16
Node Env: production
Importing Node libraries
Importing 3rd-party libraries
Importing this project modules
Prepare Notification Providers
Version: 1.14.0
Creating express and socket.io instance
Server Type: HTTP
Data Dir: ./data/
Connecting to the Database
./server/model/monitor.js is not a valid BeanModel, skipped
(node:7) Warning: Accessing non-existent property 'Proxy' of module exports inside circular dependency
(node:7) Warning: Accessing non-existent property 'prototype' of module exports inside circular dependency
SQLite config:
SQLite config:
[ { journal_mode: 'wal' } ]
[ { journal_mode: 'wal' } ]
[ { cache_size: -12000 } ]
[ { cache_size: -12000 } ]
SQLite Version: 3.36.0
Connected
SQLite Version: 3.36.0
Your database version: 10
Latest database version: 10
Database patch not needed
Database Patch 2.0 Process
[Background Job]: {
  name: 'clear-old-data',
  message: 'Clearing Data older than 10 days...'
}
[Background Job]: { name: 'clear-old-data', message: 'done' }
FATAL ERROR: Error::ThrowAsJavaScriptException napi_throw
 1: 0xb09980 node::Abort() [node]
 2: 0xa1c235 node::FatalError(char const*, char const*) [node]
 3: 0xa1c23e  [node]
 4: 0xad139b napi_fatal_error [node]
 5: 0x7f2f6539c556  [/app/node_modules/@louislam/sqlite3/lib/binding/napi-v6-linux-x64/node_sqlite3.node]
 6: 0x7f2f6539c9d8 Napi::Error::ThrowAsJavaScriptException() const [/app/node_modules/@louislam/sqlite3/lib/binding/napi-v6-linux-x64/node_sqlite3.node]
 7: 0x7f2f653ae5af  [/app/node_modules/@louislam/sqlite3/lib/binding/napi-v6-linux-x64/node_sqlite3.node]
 8: 0x7f2f653b0b34 node_sqlite3::Statement::Work_AfterPrepare(napi_env__*, napi_status, void*) [/app/node_modules/@louislam/sqlite3/lib/binding/napi-v6-linux-x64/node_sqlite3.node]
 9: 0xad38e9 node::ThreadPoolWork::ScheduleWork()::{lambda(uv_work_s*, int)#2}::_FUN(uv_work_s*, int) [node]
10: 0x155ebed  [node]
11: 0x15633c6  [node]
12: 0x1575af4  [node]
13: 0x1563d18 uv_run [node]
14: 0xa43dd5 node::SpinEventLoop(node::Environment*) [node]
15: 0xbd3262 node::worker::Worker::Run() [node]
16: 0xbd39f8  [node]
17: 0x7f2f680fcfa3  [/lib/x86_64-linux-gnu/libpthread.so.0]
18: 0x7f2f6802deff clone [/lib/x86_64-linux-gnu/libc.so.6]
```

**Comment by louislam** [comment]
Addressed the issue, it is because of circular dependency.

**Comment by louislam** [comment]
It should be fixed, please update to 1.14.1.

It is caused by bad design of the core server.

Problems:
- A lot of logic in the top level server.js
- `proxy.js` required `server.js`, since it is in another thread, despite it is not actually used, all code in server.js be executed again, then caused this issue.

To future me:
- Unless it is a very simple script, for any modules, it should put all logic in function/class. Don't put in the top level.
- The entry script never be a module, it should not required by other modules no matter what.
